src/main/java/hudson/plugins/git/extensions/impl/PruneStaleTag.java

```diff
@@ -0,0 +1,120 @@
+ * Copyright (c) 2017 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+package hudson.plugins.git.extensions.impl;
+
+import hudson.Extension;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.git.GitException;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.extensions.GitSCMExtension;
+import hudson.plugins.git.extensions.GitSCMExtensionDescriptor;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jgit.transport.RefSpec;
+import org.eclipse.jgit.transport.RemoteConfig;
+import org.eclipse.jgit.transport.URIish;
+import org.jenkinsci.plugins.gitclient.FetchCommand;
+import org.jenkinsci.plugins.gitclient.GitClient;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+
+/**
+ * Prune stale remote-tracking tags
+ *
+ * @author Nikolas Falco
+ * @since 4.2.1
+ */
+public class PruneStaleTag extends GitSCMExtension {
+    @DataBoundConstructor
+    public PruneStaleTag() {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void decorateFetchCommand(GitSCM scm,
+                                     @CheckForNull Run<?, ?> run,
+                                     GitClient git,
+                                     TaskListener listener,
+                                     FetchCommand cmd) throws IOException, InterruptedException, GitException {
+        listener.getLogger().println("Pruning obsolete local tags");
+
+        List<RemoteConfig> remoteRepos = run == null ? scm.getRepositories() : scm.getParamExpandedRepos(run, listener);
+        for (RemoteConfig remote : remoteRepos) {
+            for(URIish url : remote.getURIs()) {
+                FetchCommand fetch = git.fetch_().from(url, Arrays.asList(new RefSpec("+refs/tags/*:refs/tags/*")));
+                fetch.tags(false);
+                fetch.prune(true);
```

Comment:
Have you confirmed that this will not prune stale local branches?

It seems to be independent of the extension that prunes local branches and I assume it will not touch local branches because the refspec is specific to tags.